js/components/portal-dashboard/feedback/rubric-table.tsx

```diff
@@ -0,0 +1,155 @@
+                {this.renderRatings(crit)}
+              </div>
+            )}
+          </div>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  private renderColumnHeaders = (rubric: any) => {
+    const { referenceURL } = rubric;
+    return (
+      <div className={css.columnHeaders}>
+        <div className={css.rubricDescriptionHeader}>
+          <div className={css.scoringGuideArea}>
+            <a className={css.launchButton}>
+              <LaunchIcon className={css.icon} href={referenceURL} target="_blank" />
+            </a>
+            Scoring Guide
+        </div>
+          <div className={css.rubricDescriptionTitle}>{rubric.criteriaLabel}</div>
+        </div>
+        { rubric.ratings.map((rating: any) =>
+          <div className={css.rubricScoreHeader} key={rating.id}>
+            <div className={css.rubricScoreLevel}>{rating.label}</div>
+            {rubric.scoreUsingPoints && <div className={css.rubricScoreNumber}>({rating.score})</div>}
+          </div>
+        )
+        }
+      </div>
+    );
+  }
+
+  private renderRatings = (crit: any) => {
+    const { rubric } = this.props;
+    const { ratings } = rubric;
+    return (
+      <div id={`${crit.id}_ratings`} className={css.ratingsGroup}>
+        {ratings.map((rating: any, index: number) => this.renderStudentRating(crit, rating, index))}
+      </div>
+    );
+  }
+
+  private renderStudentRating = (crit: any, rating: any, buttonIndex: number) => {
+    const { rubricFeedback } = this.props;
+
+    const critId = crit.id;
+    const ratingId = rating.id;
+    const radioButtonKey = `${critId}-${ratingId}`;
+    const selected = (rubricFeedback && rubricFeedback[critId] && rubricFeedback[critId].id === ratingId);
+    // Tooltips displayed to teacher should actually show student description if it's available.
+    const ratingDescription =
+      (crit.ratingDescriptionsForStudent && crit.ratingDescriptionsForStudent[ratingId]) ||
+      (crit.ratingDescriptions && crit.ratingDescriptions[ratingId]) ||
+      null;
+    const isApplicableRating = crit.nonApplicableRatings === undefined ||
+      crit.nonApplicableRatings.indexOf(ratingId) < 0;
+    const isNotApplicable = !isApplicableRating;
+
+    return (
+      <div className={css.rubricScoreBox} key={radioButtonKey}>
+        <div className={css.rubricButton} title={(isApplicableRating) ? ratingDescription : "Not Applicable"}>
+          { isNotApplicable
+            ? <span className={css.noRating}>N/A</span>
+            : this.renderButton(critId, selected,ratingId, buttonIndex)
+          }
+        </div>
+      </div>
+    );
+  }
+
+  private renderButton = (critId: string, selected: boolean, ratingId: string, buttonIndex: number) => {
+    const handleRatingChange = (buttonIndex: number) => () => {
+      const { rubric, rubricFeedback } = this.props;
+      const deselect = rubric.ratings.findIndex((r: any) => r.id === (rubricFeedback[critId].id)) === buttonIndex;
```

Comment:
Might need something like this to fix the error that @emcelroy noted:
`const deselect = rubric.ratings.findIndex((r: any) => r.id === (rubricFeedback?.[critId].id)) === buttonIndex;`